src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringTests.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.PreferConstCharOverConstUnitStringAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.PreferConstCharOverConstUnitStringFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferConstCharOverConstUnitStringForStringBuilderAppendTests
+    {
+        [Fact]
+        public async Task TestRegularCase()
+        {
+            string input = @" 
+using System; 
+using System.Text;
+ 
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            StringBuilder sb = new StringBuilder();
+            const string ch = ""a"";
+            sb.Append(ch);
+        } 
+    } 
+}";
+            string fix = @" 
+using System; 
+using System.Text;
+ 
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            StringBuilder sb = new StringBuilder();
+            const char ch = 'a';
+            sb.Append(ch);
+        } 
+    } 
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(input, VerifyCS.Diagnostic(PreferConstCharOverConstUnitStringAnalyzer.Rule).WithLocation(12, 26).WithArguments("ch"), fix);
```

Comment:
You can avoid passing in explicit diagnostic instance with location and span info for verification by using markup based syntax. For example, see the markup syntax using `[|` and `|]` here: https://github.com/dotnet/roslyn-analyzers/blob/19f9e7ade9c9ec97dfb44841a6ed376df88e4296/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferTypedStringBuilderAppendOverloadsTests.cs#L28